src/group.rs

```diff
@@ -214,6 +225,71 @@ impl CanonicalDeserialize for Element {
 // Group operations
 ////////////////////////////////////////////////////////////////////////////////
 
+impl<'b> Add<&'b Element> for Element {
+    type Output = Element;
+
+    fn add(self, other: &'b Element) -> Element {
+        Element {
+            inner: self.inner + other.inner,
+        }
+    }
+}
+
+impl<'b> AddAssign<&'b Element> for Element {
+    fn add_assign(&mut self, other: &'b Element) {
+        *self = Element {
+            inner: self.inner + other.inner,
+        }
+    }
+}
+
+impl<'b> Sub<&'b Element> for Element {
+    type Output = Element;
+
+    fn sub(self, other: &'b Element) -> Element {
+        Self {
+            inner: self.inner - other.inner,
+        }
+    }
+}
+
+impl<'b> SubAssign<&'b Element> for Element {
+    fn sub_assign(&mut self, other: &'b Element) {
+        *self = Element {
+            inner: self.inner - other.inner,
+        }
+    }
+}
+
+impl Neg for Element {
+    type Output = Self;
+
+    fn neg(self) -> Self {
+        Element { inner: -self.inner }
+    }
+}
+
+impl<'b> MulAssign<&'b Scalar> for Element {
+    /// Scalar multiplication is performed through the implementation
+    /// of `MulAssign` on `EdwardsProjective` which is a type alias for
+    /// `GroupProjective<EdwardsParameters>`.
```

Comment:
I think this should probably be demoted from a doc-comment to a regular comment (either in the `mul_assign` body or above, whatever you want), since it's an implementation detail that the user of `MulAssign` shouldn't have to see in docs.

Reply:
woops yep good call